### DISCUSSION TITLE
Lazy-loaded dynamically sized grid?

### DISCUSSION THREAD

**Discussion by Absolucy** [emilk/egui#1295] [Q&A] [answered]
Title. I want to make a grid where the contents are lazily loaded (and scrollable), and the number of columns is dynamic based on window size / how many of the children can be fit into the width. Can this be done, and if so, how?

If it helps, the use-case is an online image gallery - I don't want to request all the thumbnails at once, I only want to request thumbnails and cache them as they're needed.

**Comment by emilk** [comment, marked as answer]
It is now possible with https://github.com/emilk/egui/pull/963
